### DISCUSSION TITLE
migration @bentley/imodeljs-frontend to @itwin/core-frontend (queryRealityData )

### DISCUSSION THREAD

**Discussion by PaoloPetra** [iTwin/itwinjs-core#3449] [Q&A] [answered]
Good morning everyone
I am migrating one of our apps from version 2.19 to version 3.0.2 to be aligned, and I am having a bit of difficulty

I was using @ bentley / imodeljs-frontend which has been transformed into
@ itwin / core-frontend

Inside @ bentley / imodeljs-frontend I had a "queryRealityData" method that I use to read the reality data inside my model
unfortunately with @ itwin / core-frontend this method was deleted / renamed? , how can I use queryRealityData which was very convenient for me?
otherwise should i develop an API?
as described in this link
https://developer.bentley.com/tutorials/realitydata-api-quick-start/#4-get-realitydata

Thanks for your help

**Comment by kmurari103** [comment]
Did you go through this [Reality Data](https://www.itwinjs.org/sandboxes/iTwinPlatform/Reality%20Data) sandbox, it might help you with this issue.
You can also go through the official [release note](https://www.itwinjs.org/changehistory/#:~:text=findAvailableRealityModels-,getRealityDatas,-in%20%40itwin/reality) , it has all the necessary details available.

**Comment by pmconne** [comment, marked as answer]
The [3.0.0 changelogs](https://www.itwinjs.org/changehistory/3.0.0/#itwincore-frontend) describe what was changed and how to react to those changes.
(FWIW 3.0.0 changed **a lot**. We do not ever intend to release a major version with so many breaking changes).
You can replace your usage of `queryRealityData` with the `getRealityDatas` API supplied by the [@itwin/reality-data-client package](https://www.npmjs.com/package/@itwin/reality-data-client).
(@MarcBedard8 where can we find documentation for that package's API?)
One example usage of this API can be found in [display-test-app](https://github.com/iTwin/itwinjs-core/blob/e91c4bafd6e287cdf3f2236354cd607888f20755/test-apps/display-test-app/src/frontend/ClassificationsPanel.ts#L170).

**Reply by MarcBedard8** [reply]
@EricPaquetB, Is the documentation for the [@itwin/reality-data-client package](https://www.npmjs.com/package/@itwin/reality-data-client) has been made public? Where?

**Reply by EricPaquetB** [reply]
@MarcBedard8 , we have sample code that shows how to use the reality-data-client. Unfortunately, it's not public yet, because the iTwin team is working on an authorization client for such console samples, and it's not ready yet. We need this to make our samples work.
